TESTING.md

```diff
@@ -214,6 +216,60 @@ automatically.
 * Chat responds with **potential issues** and **suggestions** like missing nullability checks or input sanitization (
   response may vary).
 
+### Edit
+
+#### Editing code
+<!-- unhide after full edit command is implemented -->
+<!-- - Verify that the option to run the `Edit Code` command is available from the sidebar, right-click menu, the command palette (option+c, /edit), or Option+K keyboard shortcut. -->
+1. Verify that the option to run the `Edit Code` command is available from the right-click menu.
+1. Highlight a section of code.
+1. Run the `Edit Code` command with instructions for how Cody should edit the selected code.
+1. Verify that Cody automatically applies edits to the selected code, per the user’s instructions.
+1. Verify that there is a notification that "Cody is working..." while edits are being applied.
+1. Verify that you can see a list of code lenses with a Cody icon above the generated code: `Show diff`, `Accept`, `Retry`, and `Undo`.
+<!-- 1. Verify that you can see a diff view of the edit in a new tab by clicking `Show diff`. -->
+1. Verify that you can prompt Cody to retry the command by clicking `Retry` and entering new instructions.
+1. Verify that you can undo the edit by clicking `Undo`.
```

Comment:
This needs to be have way more detail. For Edit & Retry, Undo and Show Diff we need to test:

- Edit the file by adding multiple lines of text before and after the region Cody is editing, while Cody is working and after it has completed, then use (Edit & Retry/Undo/Show Diff)
- Edit the file by *removing* multiple lines of text before and after the region Cody is editing, while Cody is working and after it has completed, then use (Edit & Retry/Undo/Show Diff)
- Edit the file by *removing* multiple lines of text before and *adding multiple lines of text after* the region Cody is editing, while Cody is working and after it has completed, then use (Edit & Retry/Undo/Show Diff)
- When testing Undo specifically, only the changes Cody made should be undone.
- When testing Show Diff specifically, only the changes in the region Cody edited should be highlighted in the diff.

You need a preamble before ALL of these commands which says "when we say CONCURRENT HUMAN EDIT, we mean: no editing, add lines of text before the region Cody is editing, ..." and then in the test plan, you need to say "For each kind of CONCURRENT HUMAN EDIT: ..." and then have the list of instructions with one level of indent.